android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesExecutor.kt

```diff
@@ -80,26 +81,24 @@ class RulesExecutor @Inject constructor(val rulesFactory: RulesFactory) {
 
   private suspend fun computeRules(
     ruleConfigs: List<RuleConfig>,
-    ruleConfigsKey: String,
     baseResource: Resource,
     relatedResourcesMap: Map<String, List<Resource>>
   ): Map<String, Any> =
     // Compute values via rules engine and return a map. Rule names MUST be unique
     rulesFactory.fireRules(
-      rules = rulesFactory.generateRules(ruleConfigsKey, ruleConfigs),
+      rules = rulesFactory.generateRules(ruleConfigs),
       baseResource = baseResource,
       relatedResourcesMap = relatedResourcesMap
     )
 
   private suspend fun List<Resource>.mapToResourceData(
     relatedResourcesMap: MutableMap<String, MutableList<Resource>>,
     ruleConfigs: List<RuleConfig>,
-    ruleConfigsKey: String,
     listRelatedResources: List<ExtractedResource>,
     computedValuesMap: Map<String, Any>
   ) =
     this.map { resource ->
-      val start = System.currentTimeMillis()
+      val start0 = System.currentTimeMillis()
```

Comment:
You can use Kotlin's lambda function [measureTimeMillis](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.system/measure-time-millis.html)